pontoon/tags/utils/resources.py

```diff
@@ -0,0 +1,108 @@
+
+
+class TagsResourcesTool(TagsDataTool):
+    filter_methods = ('locales', 'projects', 'slug', 'path')
+
+    @property
+    def data_manager(self):
+        return self.resource_manager
+
+    def link(self, tag, glob=None, resources=None):
+        tag = self.tag_manager.get(slug=tag)
+        if glob is not None:
+            resources = list(self.find(glob, exclude=tag))
+            for resource in resources:
+                self._validate_resource(tag, resource.project_id)
+            tag.resources.add(*resources)
+            return resources
+        elif resources is not None:
+            _resources = self.resource_manager.none()
+            for resource in resources:
+                self._validate_resource(tag, resource['project'])
+                _resources |= self.resource_manager.filter(
+                    project=resource["project"],
+                    path=resource["path"])
+            tag.resources.add(*list(_resources))
+
+    @property
+    def filtered_data(self):
+        return super(TagsResourcesTool, self).filtered_data.distinct()
+
+    def filter_locales(self, resources):
+        return (
+            resources.filter(translatedresources__locale__in=self.locales)
+            if self.locales
+            else resources)
+
+    def filter_path(self, resources):
+        return (
+            resources.filter(path__regex=glob_to_regex(self.path))
+            if self.path
+            else resources)
+
+    def filter_projects(self, resources):
+        return (
+            resources.filter(project__in=self.projects)
+            if self.projects
+            else resources)
+
+    def filter_slug(self, resources):
+        return (
+            resources.filter(tag__slug__regex=glob_to_regex(self.slug))
+            if self.slug
+            else resources)
+
+    def find(self, glob, include=None, exclude=None):
+        if include:
+            resources = self.resource_manager.filter(tag__slug=include)
+        elif exclude:
+            resources = self.resource_manager.exclude(tag__slug=exclude)
+        else:
+            resources = self.resource_manager
+        if self.projects:
+            resources = resources.filter(project__in=self.projects)
+        return resources.filter(path__regex=glob_to_regex(glob))
+
+    def get(self, tag):
+        return self.filtered_data.filter(tag__slug=tag).distinct()
+
+    def get_linkable_resources(self, slug):
+        return self.find('*', exclude=slug).values('path', 'project')
+
+    def get_linked_resources(self, slug):
+        return self.get(slug).values('path', 'project')
+
+    def get_linked_locales(self, slug):
+        return self.locale_manager.filter(
+            pk__in=self.get(slug).values_list(
+                "translatedresources__locale").distinct())
+
+    def get_linked_projects(self, slug):
+        return self.project_manager.filter(
+            pk__in=self.get(slug).values_list(
+                "project").distinct())
```

Comment:
There's an inconsistency between `get_linked_resources` vs. `get_linked_locales` and `get_linked_projects` in that the former returns `values` and the latter two return `values_list`.

Reply:
ill update the the latter 2 to return values

Reply:
in the end i removed the latter 2 methods - they were part of the site-wide tags implementation, and are not required/used - so better remove, at least for now